vm/array_test.go

```diff
@@ -155,6 +156,110 @@ func TestEachMethod(t *testing.T) {
 	}
 }
 
+func TestMapMethod(t *testing.T) {
+	//generate array of ints
+	var elementsInt []Object
+	for i := 1; i <= 5; i++ {
+		int := initilaizeInteger(i + 3)
+		elementsInt = append(elementsInt, int)
+	}
+
+	//generate array of booleans
+	var elementsBool []Object
+	for i := 1; i <= 5; i++ {
+		b := initializeBoolean(i%2 == 0)
+		elementsBool = append(elementsBool, b)
+	}
+
+	//generate array of strings
+	var elementsStr []Object
+	for i := 1; i <= 5; i++ {
+		s := initializeString(strconv.Itoa(i) + "1")
+		elementsStr = append(elementsStr, s)
+	}
+
+	tests := []struct {
+		input    string
+		expected *ArrayObject
+	}{
+		{`
+		a = [1, 2, 3, 4, 5]
+		a.map do |i|
+			i + 3
+		end
+		`, InitializeArray(elementsInt)},
+		{`
+		a = [true, false, true, false, true ]
+		a.map do |i|
+			!i
+		end
+		`, InitializeArray(elementsBool)},
+		{`
+		a = ["1", "2", "3", "4", "5"]
+		a.map do |i|
+			i + "1"
+		end
+		`, InitializeArray(elementsStr)},
+	}
+
+	for _, tt := range tests {
+		evaluated := testEval(t, tt.input)
+		testArrayObject(t, evaluated, tt.expected)
+	}
+}
+
+func TestSelectMethod(t *testing.T) {
+	//generate array of ints
+	var elementsInt []Object
+	for i := 4; i <= 5; i++ {
+		int := initilaizeInteger(i)
+		elementsInt = append(elementsInt, int)
+	}
+
+	//generate array of booleans
+	var elementsBool []Object
+	for i := 1; i <= 3; i++ {
+		b := initializeBoolean(true)
+		elementsBool = append(elementsBool, b)
+	}
+
+	//generate array of strings
+	var elementsStr []Object
+	for i := 1; i <= 2; i++ {
+		s := initializeString("test")
+		elementsStr = append(elementsStr, s)
+	}
+
+	tests := []struct {
+		input    string
+		expected *ArrayObject
+	}{
+		{`
+		a = [1, 2, 3, 4, 5]
+		a.select do |i|
+			i > 3
+		end
+		`, InitializeArray(elementsInt)},
```

Comment:
I think you should generate array with clear arguments. For example in this case it should be
```go
InitializeArray(initilaizeInteger(4), initilaizeInteger(5))
```
I know this seems very verbose but I can now know the expected result just by watching this case without scrolling. And this can be further refactored in the future. Same rule applies to other tests.